# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/spellchecker/TextServicesDocument.h
# Commit: 622d9feefc23
# Full Hash: 622d9feefc23b5cc1536082fbcb99231d1f43a8f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-07-13 21:42:32
# Regressor Bug: 1718924
# File Overlap Count: 2
# Description:
#   Bug 1718924 - part 12: Move `TextServicesDocument::FindWordBounds()` to `OffsetEntryArray` r=m_kato
#   
#   Depends on D119158
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119159
# ==============================================================================

diff -r a5c5378612e0 -r 622d9feefc23 editor/spellchecker/TextServicesDocument.h
--- a/editor/spellchecker/TextServicesDocument.h	Tue Jul 13 10:17:48 2021 +0000
+++ b/editor/spellchecker/TextServicesDocument.h	Tue Jul 13 10:19:40 2021 +0000
@@ -6,6 +6,7 @@
 #ifndef mozilla_TextServicesDocument_h
 #define mozilla_TextServicesDocument_h
 
+#include "mozilla/EditorDOMPoint.h"
 #include "mozilla/Maybe.h"
 #include "mozilla/UniquePtr.h"
 #include "nsCOMPtr.h"
@@ -64,6 +65,13 @@
     Result<IteratorStatus, nsresult> Init(
         FilteredContentIterator& aFilteredIter, IteratorStatus aIteratorStatus,
         nsRange* aIterRange, nsAString* aAllTextInBlock = nullptr);
+
+    /**
+     * FindWordRange() returns a word range starting from aStartPointToScan
+     * in aAllTextInBlock.
+     */
+    Result<EditorDOMRangeInTexts, nsresult> FindWordRange(
+        nsAString& aAllTextInBlock, const EditorRawDOMPoint& aStartPointToScan);
   };
 
   RefPtr<dom::Document> mDocument;
@@ -315,13 +323,6 @@
 
   nsresult RemoveInvalidOffsetEntries();
   nsresult SplitOffsetEntry(size_t aTableIndex, uint32_t aOffsetIntoEntry);
-
-  static nsresult FindWordBounds(nsTArray<UniquePtr<OffsetEntry>>* aOffsetTable,
-                                 nsString* aBlockStr, nsINode* aNode,
-                                 uint32_t aNodeOffset, nsINode** aWordStartNode,
-                                 uint32_t* aWordStartOffset,
-                                 nsINode** aWordEndNode,
-                                 uint32_t* aWordEndOffset);
 };
 
 }  // namespace mozilla
